nuxeo-features/rest-api/nuxeo-rest-api-test/src/test/java/org/nuxeo/ecm/restapi/test/DocumentBrowsingTest.java

```diff
@@ -487,10 +489,85 @@ public void itCanSetArrayPropertyToNull() throws IOException {
             // Then the document is updated
             fetchInvalidations();
             doc = session.getDocument(doc.getRef());
+            // ArrayProperty returns null whenever is empty or null
             assertNull(doc.getPropertyValue("dc:subjects"));
         }
     }
 
+    /*
+     * NXP-28433
+     */
+    @Test
+    @Deploy("org.nuxeo.ecm.platform.restapi.server:test-defaultvalue-docTypes.xml")
+    public void itCanSetArrayPropertyOfComplexToEmpty() throws IOException {
+        DocumentModel doc = session.createDocumentModel("/", "myDocument", "DocDefaultValue");
+        Map<String, String> complex = new HashMap<>();
+        complex.put("foo", "val1");
+        complex.put("bar", "val2");
+        doc.setProperty("defaultvalue", "multiComplexWithoutDefault", Collections.singletonList(complex));
+        doc = session.createDocument(doc);
```

Comment:
i don't remember if we privilege the using of `doc.setProperty()` instead of `doc.setPropertyValue()` i asked this question because i think we can replace with: 
```
 doc.setPropertyValue("dv:complexWithoutDefault/foo", "val1");
        doc.setPropertyValue("dv:complexWithoutDefault/bar", "val2");
```

as they did on the previous tests instead of the creation `Map`...

Reply:
This is not the same type than the example you pasted. Here we have a list of complex, I use the setProperty method because other one needs a Serializable which force us to cast the thing, I don't like that.